astronomer/providers/google/cloud/operators/dataproc.py

```diff
@@ -190,22 +190,29 @@ def execute(self, context: Context) -> None:
             metadata=self.metadata,
         )
 
-        end_time: float = time.time() + self.timeout
+        try:
+            hook.get_cluster(project_id=self.project_id, region=self.region, cluster_name=self.cluster_name)
+        except NotFound:
+            return
+        except Exception as e:
+            raise AirflowException(str(e))
+        else:
```

Comment:
Do we need this else block? If any of the exception occurs the flow will be terminated anyway and it won't reach here. My suggestion is to remove this additional nesting for the new lines if we don't need it.

Reply:
I just removed it. Thanks!